ycmd/completers/python/jedi_completer.py

```diff
@@ -46,24 +62,111 @@ def SupportedFiletypes( self ):
     return [ 'python' ]
 
 
-  def _GetJediScript( self, request_data ):
-      filename = request_data[ 'filepath' ]
-      contents = request_data[ 'file_data' ][ filename ][ 'contents' ]
-      line = request_data[ 'line_num' ]
-      # Jedi expects columns to start at 0, not 1
-      column = request_data[ 'column_num' ] - 1
+  def _Shutdown( self ):
+    if self.ServerIsRunning():
+      self._StopServer()
+
+
+  def ServerIsReady( self ):
+    """ Check if JediHTTP server is ready. """
+    try:
+      return bool( self._GetResponse( '/ready' ) )
+    except:
```

Comment:
I'm not sure if you're aware or not, but usually you don't want a "bare" `except:`.

>A bare `except:` clause will catch SystemExit and KeyboardInterrupt exceptions, making it harder to interrupt a program with Control-C, and can disguise other problems. If you want to catch all exceptions that signal program errors, use `except Exception:`.

https://www.python.org/dev/peps/pep-0008/#programming-recommendations